hack/util.sh

```diff
@@ -426,6 +426,28 @@ function util::kubectl_with_retry() {
     return ${ret}
 }
 
+# util::delete_all_clusters deletes all clusters directly
+# util::delete_all_clusters actually do three things: delete cluster、remove kubeconfig、record delete log
+# Parmeters:
+#  - $1: KUBECONFIG file of host cluster, such as "~/.kube/karmada.config"
+#  - $2: KUBECONFIG file of member cluster, such as "~/.kube/members.config"
+#  - $3: log file path, such as "/tmp/karmada/"
+function util::delete_all_clusters() {
+  local main_config=${1}
+  local member_config=${2}
+  local log_path=${3}
+
+  local log_file="${log_path}"/delete-all-clusters.log
+  rm -rf ${log_file}
+  mkdir -p ${log_path}
+
+  kind delete clusters --all >> "${log_file}" 2>&1
+  rm -f "${main_config}"
+  rm -f "${member_config}"
```

Comment:
It just feels a little bit rough to delete the entire kubeconfig file.
If the file contains a cluster not created by kind, the configuration will be removed unexpectedly. 

It's not a big deal though.

Reply:
Generally LGTM. It seems that the [logic for removing kubeconfig](https://github.com/karmada-io/karmada/blob/master/hack/local-down-karmada.sh#L37-L50) is included here. And the [logic for removing kind clusters](https://github.com/karmada-io/karmada/blob/master/hack/local-down-karmada.sh#L29-L35) scope is expanded. 

I think it's more suitable to just delete kind clusters which created by `local-up-karmada.sh`.

Reply:
> It just feels a little bit rough to delete the entire kubeconfig file. If the file contains a cluster not created by kind, the configuration will be removed unexpectedly.
> 
> It's not a big deal though.

Although `kind delete` is capable of cleaning kubeconfig if specificed `$KUBECONFIG` or `--kubeconfig`, it is not as effective as expected. I have tried these two modification method:

1. one

```shell
export KUBECONFIG="${main_config}":"${member_config}"
kind delete clusters --all >> "${log_file}" 2>&1
unset KUBECONFIG
```

2. two

```shell
kind delete clusters --kubeconfig "${main_config}" karmada-host
kind delete clusters --kubeconfig "${member_config}" member1 member2 member3
```

All above methods can not clean kubeconfig content thoroughly.

***

By the way, even if no `rm -f xxx_config` in `delete_all_clusters`, following the original logic, the script will eventually execute `rm -f xxx_config` in `util::create_cluster`. 

the code of `util::create_cluster`:

```bash
function util::create_cluster() {
  local cluster_name=${1}
  local kubeconfig=${2}
  local kind_image=${3}
  local log_path=${4}
  local cluster_config=${5:-}

  mkdir -p ${log_path}
  rm -rf "${log_path}/${cluster_name}.log"
  rm -f "${kubeconfig}"
  ...
}
```

Reply:
OK. got it.